lib/dispatcher/client-h1.js

```diff
@@ -182,10 +185,17 @@ class Parser {
       // istanbul ignore else: only for jest
       if (this.timeout.refresh) {
         this.timeout.refresh()
+        this.timeoutDeadline = this.timeoutValue ? Date.now() + this.timeoutValue : 0
       }
     }
   }
 
+  updateTimeout () {
+    if (this.timeoutDeadline && this.timeoutDeadline < Date.now()) {
+      onParserTimeout(this)
+    }
```

Comment:
This will call `Date.now()` quite often... which is not optimal performance-wise.